package/base-files/files/lib/upgrade/stage2

```diff
@@ -95,27 +95,25 @@ kill_remaining() { # [ <signal> [ <loop> ] ]
 			# Skip kernel threads
 			[ -n "$cmdline" ] || continue
 
-			_vn " $name"
+			v "Sending signal $sig to $name ($pid)"
 			kill -$sig $pid 2>/dev/null
 
 			[ $loop -eq 1 ] && run=true
 		done
 
 		let loop_limit--
 		[ $loop_limit -eq 0 ] && {
-			_v
 			v "Failed to kill all processes."
 			exit 1
 		}
 	done
-	_v
 }
 
 indicate_upgrade
 
-killall -9 telnetd
-killall -9 dropbear
-killall -9 ash
+pgrep telnetd >/dev/null && killall -9 telnetd
+pgrep dropbear >/dev/null && killall -9 dropbear
+pgrep ash >/dev/null && killall -9 ash
```

Comment:
Redirecting stderr to /dev/null should suffice.  There exists a race condition between `pgrep` and `killall`

Reply:
@yousong I can do that too, of course.
The question that comes to my mind is, why do we make a difference her?
Especially telnet is not installed on my targets anymore.
If we redirect it now, we no longer get to see it when we sort out these packages in the future.
Can we not dispense with this distinction altogether?

Reply:
> The question that comes to my mind is, why do we make a difference her?
> Especially telnet is not installed on my targets anymore.
> If we redirect it now, we no longer get to see it when we sort out these packages in the future.
> Can we not dispense with this distinction altogether?

I traced those sigkill lines to https://github.com/openwrt/openwrt/commit/1e876b4191ea0d159ec74087478e1ef30a6bb5d0 in 2014.  @blogic may help here.

I can only guess that the hard termination of these 3 kinds of process are for minimalising possible outside interferences.

Reply:
@yousong I have now redirected the output for ash, dropbear and telnetd to /dev/null as you suggested.